### ISSUE TITLE
Crate of the week for TWIR '95

### ISSUE THREAD

**Issue by llogiq** [cmr/this-week-in-rust#111]
Something like the following:

----
This is a new part of this weekly installation, where [llogiq](https://github.com/llogiq) will write about a crate that some of you may not know. Please nominate a crate of your choice at the [rust-users thread](https://users.rust-lang.org/t/crate-of-the-week/2704/15) so he can write about it next week.

For this first installment, the most votes went to [serde](https://github.com/serde-rs/serde). Despite many of you probably knowing it, I accepted the popular vote because of the great benefit those who don't yet know it may get. Thanks go to [bstrie](https://users.rust-lang.org/users/bstrie) for the suggestion.

Serde (de)serializes arbitrary Rust data to a number of formats, including [json](https://crates.io/crates/serde_json), [XML](https://github.com/serde-rs/xml), [YAML](https://github.com/serde-rs/yaml/), [Bincode](https://crates.io/crates/bincode) and [MessagePack](https://crates.io/crates/rmp). It is possible to write encoder/decoder pairs to work with Serde that don't need to know too much about Rust objects, and Rust data structures only need a pair of auto-derived traits to work with Serde. Now what can one use serializaton for? Storing rust objects in files or database blobs, sending them between processes, over language-barriers or over the network, for one.

In short, serde is *the* rust framework for your (de)serialization needs.

**Comment by llogiq** [comment]
@nasa42 @brson Are you OK with this text? Do you have suggestions for improving it?

**Comment by nasa42** [comment]
Looks great! Though the first paragraph can be modified a bit to reflect that this a team effort rather than of one individual (replacing _he_ with _we_ etc.)? Or replace it entirely and just mention the TRPLF thread?
We do have a line in the draft to mention all contributors for this week's issue (though I never used it when I'm the only author). :-)

**Comment by llogiq** [comment]
Ok, so replace the first two paragraphs with:
----
This is a new part of this weekly installation, where we will write about a crate that some of you may not know. Please nominate a crate of your choice at the [rust-users thread](https://users.rust-lang.org/t/crate-of-the-week/2704/15) so we can write about it next week.

For this first installment, the most votes went to [serde](https://github.com/serde-rs/serde). Despite many of you probably knowing it, we accepted the popular vote because of the great benefit those who don't yet know it may get. Thanks go to [bstrie](https://users.rust-lang.org/users/bstrie) for the suggestion.

**Comment by nasa42** [comment]
Fixed capitalisation and added the section in #112. This can be closed now. Thanks @llogiq!

**Comment by brson** [comment]
Awesome!

**Comment by llogiq** [comment]
Thanks @nasa42! And let's see what we get next week.
